pkg/controller/jobframework/reconciler.go

```diff
@@ -1148,24 +1150,62 @@ func (r *JobReconciler) ignoreUnretryableError(log logr.Logger, err error) error
 	return err
 }
 
-func generatePodsReadyCondition(job GenericJob, wl *kueue.Workload) metav1.Condition {
-	conditionStatus := metav1.ConditionFalse
-	message := "Not all pods are ready or succeeded"
-	// Once PodsReady=True it stays as long as the workload remains admitted to
-	// avoid unnecessary flickering the condition when the pods transition
-	// from Ready to Completed. As pods finish, they transition first into the
-	// uncountedTerminatedPods staging area, before passing to the
-	// succeeded/failed counters.
-	if workload.IsAdmitted(wl) && (job.PodsReady() || apimeta.IsStatusConditionTrue(wl.Status.Conditions, kueue.WorkloadPodsReady)) {
-		conditionStatus = metav1.ConditionTrue
-		message = "All pods were ready or succeeded since the workload admission"
-	}
-	return metav1.Condition{
-		Type:    kueue.WorkloadPodsReady,
-		Status:  conditionStatus,
-		Reason:  "PodsReady",
-		Message: message,
-		// ObservedGeneration is added via workload.UpdateStatus
+func generatePodsReadyCondition(log logr.Logger, job GenericJob, wl *kueue.Workload) metav1.Condition {
+	if !workload.IsAdmitted(wl) {
+		// The workload has not been admitted yet
+		// or it was admitted in the past but it's evicted/requeued
+		return workload.CreatePodsReadyCondition(metav1.ConditionFalse,
+			kueue.WorkloadNotAdmitted,
+			"Workload isn't admitted")
+	}
```

Comment:
I do not think that this is needed since we will do as following, IIUC:

- `it was admitted in the past but it's evicted/requeued`: kueue turns the QuotaReserved condition to False and removes Admitted condition from Workload.
- `The workload has not been admitted yet`: We tell by the `Admitted` and `QuotaReserved` conditions whether or not the Workload is admitted. What is the value of an additional admitted state rather than `Admitted` and `QuotaReserved`.

Reply:
My primary motivation was that after creating a Workload we set the PodsReady condition to False, even tho it hasn't been admitted yet, and is not counting down. I thought it can be misleading so I wanted to differentiate the case when the workload is not admitted yet, and when it's admitted and counting down

Reply:
In that case could we clear the `PodsReady` condition if the Workload has it since `{Admitted: nil, QuotaReserved: false, PodsReady: false}` is weird.

The `{QuotaReserved: False}` covers all cases for `WorkloadNotAdmitted`. I mean we represent non admitted workloads by `{QuotaRerserved: False}` and `WorkloadNonAdmitted` reason. I think we should represent the state by only one semantics.

Do you have any logic to verify something by this `WorkloadNonAdmitted` reason?

Reply:
> In that case could we clear the `PodsReady` condition if the Workload has it since `{Admitted: nil, QuotaReserved: false, PodsReady: false}` is weird.

Just so you know, this is the current Kueue's behavior. 

Could you elaborate on what you mean by "clearing" the condition? Do you mean to set empty strings in Condition's Status, Reason and Message?

Reply:
> After this PR, `Admitted: True` or `PodsReady: False, Reason: NonAdmitted` can represent admission status.

I see, that's a good catch.

> In conclusion, basically, I think we should represent Admission state only by Admitted and QuotaReserved.

@mimowo are you fine with getting rid of the `NotAdmitted` reason and sticking with the 1st approach discussed [here](https://github.com/kubernetes-sigs/kueue/pull/4301#discussion_r1967304906)?

Reply:
The NotAdmitted reason is not a new way to represent not admitted workload. This remains solely dependent on the Admitted condition type.

The reason is only used within the context of the PodsReady condition to indicate why PodsReady is False.

That being said I'm ok to not introduce the reason as part of this PR because it is a bit orthogonal and it wasn't there ok. I still think it makes sense as a follow up

Reply:
SGTM, I'll revert the changes related to NotAdmitted, and will create a new issue once this PR is merged